### PR TITLE
Feature/workallocation 5

### PR DESCRIPTION
**Pull request by udaydenduluri33**
Its a feature toggled Work Allocation work for the last 4 sprints. This supersedes workallocation-4 branch. 
Each and every commit/PR that came to this branch is code reviewed

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: src/app/components/hmcts-global-header/hmcts-global-header.component.ts [hunk 1]
```diff
@@ -25,14 +26,54 @@ export class HmctsGlobalHeaderComponent {
...
+        return 'Notice of change';
+      }
+      default: {
+        return null;
+      }
+    }
+  }
 
   public onEmitEvent(index: number): void {
+    console.log('here')
```
Comment (paul-graham) [inline src/app/components/hmcts-global-header/hmcts-global-header.component.ts:69]: This should be removed.

File: src/app/components/hmcts-global-header/hmcts-global-header.component.ts [hunk 2]
```diff
@@ -25,14 +26,54 @@ export class HmctsGlobalHeaderComponent {
...
     this.navigate.emit(this.navigation.items[index].emit);
   }
 
   public onEmitSubMenu(menuItem: any) {
+    // remove the setting of selected item via url
+    this.tab = menuItem.text;
+    console.log(menuItem.text);
```
Comment (paul-graham) [inline src/app/components/hmcts-global-header/hmcts-global-header.component.ts:76]: This should be removed.

**Review by paul-graham — CHANGES_REQUESTED** [review]

File: src/cases/containers/case-details-home/case-details-home.component.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    if (navigation) {
+      this.extras = navigation.extras;
+    }
+  }
+
+  public ngOnInit() {
+    if (this.extras && this.extras.state && this.extras.state.showMessage && this.extras.state.messageText) {
+      this.alertService.success(this.extras.state.messageText);
+    }
+    this.activatedRoute.data.subscribe(data => console.log('Activated router data is', data));
```
Comment (paul-graham) [inline src/cases/containers/case-details-home/case-details-home.component.ts:28]: Don't need this `console.log()`.

File: src/work-allocation/containers/available-tasks/available-tasks.component.ts [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        this.refreshTasks();
+      }
+    }
+  }
+
+  /**
+   * Handle a User Claiming a Task
+   */
+  public onActionHandler(taskAction: InvokedTaskAction): void {
+    console.log(taskAction.action.id);
```
Comment (paul-graham) [inline src/work-allocation/containers/available-tasks/available-tasks.component.ts:125]: Don't need this `console.log()`.

**Review by paul-graham — APPROVED** [review]

File: src/cases/containers/case-details/case-details.component.spec.ts [hunk 5]
```diff
@@ -1,37 +1,38 @@
...
+
 class MockSortService {
-  features = {};
-  getFeatureToggle() { }
-  getEditorConfiguration() { }
+  public features = {};
+  public getFeatureToggle() { }
+  public cgetEditorConfiguration() { }
 }
 xdescribe('CaseDetailsComponent', () => {
```
Comment (connorpgpmcelroy) [inline src/cases/containers/case-details/case-details.component.spec.ts:37]: What is the point of this spec file if we have an xdescribe? Adnan mentioned on other PR that we should be getting rid of these so is this going to be implemented in future?

File: src/cases/containers/index.ts [hunk 6]
```diff
@@ -5,7 +5,7 @@ import {CaseSearchComponent} from './case-search/case-search.component';
 import {CasesCreateComponent} from './case-create/case-create.component';
 import {CaseDetailsComponent} from './case-details/case-details.component';
 import { CaseCreateSubmitComponent } from './case-create-submit/case-create-submit.component';
-import {CaseDetailsHomeComponent} from './case-details-home/case-details-home';
+import {CaseDetailsHomeComponent} from './case-details-home/case-details-home.component';
```
Comment (connorpgpmcelroy) [inline src/cases/containers/index.ts:8]: Inconsistent formatting. I could quickly change this?

File: src/work-allocation/models/comms/infomation-message.model.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+import {InfoMessage, InfoMessageType} from './../../enums';
```
Comment (connorpgpmcelroy) [inline src/work-allocation/models/comms/infomation-message.model.ts:1]: Should this be spaced within brackets for consistency?

**Review by connorpgpmcelroy — COMMENTED** [review]
Commented with some queries

File: src/cases/containers/case-details/case-details.component.spec.ts (hunk 5 above)
Comment (udaydenduluri33) [reply src/cases/containers/case-details/case-details.component.spec.ts:37]: @connorpgpmcelroy  its a good practice to remove it. But, not essential for this PR.

File: src/cases/containers/index.ts (hunk 6 above)
Comment (udaydenduluri33) [reply src/cases/containers/index.ts:8]: please do it, Thanks

**Review by connorpgpmcelroy — APPROVED** [review]
LGTM

**Pull request merged**